torchgeo/datasets/l7irish.py

```diff
@@ -0,0 +1,253 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Landsat 7 Cloud Cover Assessment Validation Data"""
+import abc
+import glob
+import hashlib
+import os
+from functools import lru_cache
+from typing import Any, Callable, Dict, List, Optional, cast
+
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from matplotlib.colors import ListedColormap
+from PIL import Image
+from rasterio.crs import CRS
+from torch import Tensor
+from torch.utils.data import Dataset
+
+from .geo import NonGeoDataset, RasterDataset
+from .landsat import Landsat7
+from .utils import BoundingBox, download_url, extract_archive, working_dir
+
+
+class Landsat7Irish(RasterDataset):
+    
+    """ Landsat 7 Cloud Cover Assessment Validation Data.
+
+    The `Landsat 7 Cloud Cover Assessment Validation Data <https://landsat.usgs.gov/landsat-7-cloud-cover-assessment-validation-data>`__ dataset is based on Landsat 7
+    Enhanced Thematic Mapper Plus (ETM+) Level-1G scenes, displayed in the biomes listed below. Manually generated cloud masks are used to train and validate cloud cover 
+    assessment algorithms, which in turn are intended to compute the percentage of cloud cover in each scene. *Originally, this was reported as 207 masks, but one mask 
+    was erroneously counted.
+
+    Dataset format:
+    * 102 scenes from this dataset were used in a cloud validation study described in Foga, Scaramuzza et al. (2017)
```

Comment:
This sentence has nothing to do with the dataset format, you should instead list # images, image size, # spectral bands, Level-1G, etc.